model/godleyTableWindow.cc

```diff
@@ -798,6 +798,55 @@ namespace minsky
       godleyIcon->deleteRow(r+1);
     requestRedraw();
   }
+  
+  string GodleyTableWindow::moveAssetClass(double x, double y)
+  {
+	x/=zoomFactor;
+	y/=zoomFactor;
+	unsigned c=colX(x);	
+	string tmpStr="";
+	if (clickType(x,y)==colWidget) {
+	    unsigned visibleCol=c-scrollColStart+1;
+        if (c<colWidgets.size() && visibleCol < colLeftMargin.size()) {
+		    auto& moveVar=godleyIcon->table.cell(0,c);
+		    auto oldAssetClass=classdesc::enumKey<GodleyAssetClass::AssetClass>(godleyIcon->table._assetClass(c));			
+            if (colWidgets[c].button(x-colLeftMargin[visibleCol])==3) {
+				auto targetAssetClass=classdesc::enumKey<GodleyAssetClass::AssetClass>(godleyIcon->table._assetClass(c+1));
+				if (targetAssetClass!=oldAssetClass && !moveVar.empty())
+					tmpStr="This will convert "+moveVar+" from "+oldAssetClass+" to "+targetAssetClass+". Are you sure?";
```

Comment:
These two lines of code have effectively been repeated 4 times, and are a candidate for extraction into a function.

Reply:
Definitely agree. I wanted to do it "in place" but with the changing values of oldAssetClass and targetAssetClass that wouldn't work. But a function is the obvious alternative. Thanks for the suggestion :)